ocaml/idl/datamodel_types.ml

```diff
@@ -26,12 +26,12 @@
 
 *)
 module Date = struct
-  open Xapi_stdext_date
+  module Date = Xapi_stdext_date.Date
   include Date
 
-  let iso8601_of_rpc rpc = Date.of_string (Rpc.string_of_rpc rpc)
+  let t_of_rpc rpc = Date.of_iso8601 (Rpc.string_of_rpc rpc)
```

Comment:
These functions need to be declared because of the ppx: previously the type used was `iso8601`, now it's `t`. This means that the functions need to change the name and match the type's name.
On top of that, the conversion function has changed name to make explicit the format of the string, which corresponds to the name of the previous type name.